### PR TITLE
Project initialization

### PR DESCRIPTION
**Pull request by FledgeXu**
## Goal
Relates: #1 
The structure of this project is initialized in this PR.
## Brief
The project is divided into two parts: backend and frontend. 

The backend part is a FastAPI project and uses pipenv as package manager.(followed  the suggestion of [overview#16](https://github.com/openzim/overview/issues/16)). All the tasks(like format, tests) would be managed by the `invoke`, so we don't have to maintain package version consistency in multiple places.

The frontend part is a default Vue project. The only thing worth noticing is that Vue uses `vite` as the default build tool instead of `webpack` from Vue3.

### CODE REVIEW

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thanks ; it looks good ; I don't have much to say.

On the backend part, given https://github.com/openzim/overview/issues/16 is not decided yet and because it seems to be more and more widespread, I'd prefer we use a `pyproject.toml` file for dependencies and metadata. I believe pipenv can work with it.

As for the frontend, I read the Vite arguments on their website and it feels quite speedy indeed for development.

Regarding Pinia, I understand it's an alternative to VueX ? I've skimmed through the docs and it seems indeed more accessible which is a good point. It also became the [official store lib](https://vuejs.org/guide/scaling-up/state-management.html#pinia) so 👍 

In the future, please add this kind of details to the PR as we obviously don't follow the evolution of JS frameworks (nor python ones!) and it's important we make informed choice because we try to have a consistent code base to ease maintenance over projects.

Please use the pyproject.toml and we'll merge this

File: .github/workflows/qa.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          pip install --upgrade pipenv
+          pipenv sync --dev
+
+      - name: Check black formatting
+        working-directory: backend
+        run: pipenv run invoke check-black
+
+      - name: Check import order with isort
+        working-directory: backend
+        run: pipenv run invoke check-isort
```
[the remark this answers is not in the document]
Comment (FledgeXu) [reply .github/workflows/qa.yaml:29]: I misunderstood it. I thought that flake8 only check code style but actually, it's not. I will add that then.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          pip install --upgrade pipenv
+          pipenv sync --dev
+
+      - name: Run tests
+        working-directory: backend
+        run: |
+          pipenv run invoke test --args "-v"
+          pipenv run invoke generate-xml-report
+          pwd
+
```
[the remark this answers is not in the document]
Comment (FledgeXu) [reply .github/workflows/test.yaml:29]: I saw this part, but I don't know the token of cododev, Should I apply for one or something else?

File: .github/workflows/qa.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgaudin) [reply .github/workflows/qa.yaml:29]: Would be in addition to black and isort which are formatters. But we need the flake8 checks ; very useful.
Make sure to specify the max-line-length as we want to stick with the black version of `88`.

File: backend/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+src/coverage.xml
```
[the remark this answers is not in the document]
Comment (FledgeXu) [reply backend/.gitignore:1]: All of the temporary files of the Python project are ignored in the .gitignore of the root folder. I will move that part to this .gitignore.

File: frontend/.gitignore [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Logs
```
[the remark this answers is not in the document]
Comment (FledgeXu) [reply frontend/.gitignore:1]: Default `.gitignore` is good enough.

**Comment by FledgeXu** [comment]
Update:
1. Moved the dependency and metadata sections to the pyproject.toml file.
2. Add flake8 check.
3. Add codedev support.
4. Adjust `.gitignore`.
5. tweak the `tasks.py` scripts section of `Pipfile` and CI.

**Review by rgaudin — APPROVED** [review]
Thank you ; I think we're good to go.

I'd prefer we include the link to the URL on gitignore.io that produced the ignore (for each) so it facilitates maintenance in the future.

Also I have a very different `yarn.lock` because I'm on a different architecture.
I also have a, not-on-repo but not ignored `package-lock.json` in frontend.
I believe this could be solved by updating the documentation to use `yarn install --frozen-lockfile` instead of `npm install` as yarn resolver is more predictable and this option would not modify the lock file.

Finally, I noticed you forced-push at some point. Try to not force-push nor change any already-pushed commit in between the time you request a review and the time it is accepted. Can lead to very frustrating experience.
We do the potential fixup/cleanup post-acceptance and before merge.

**Comment by FledgeXu** [comment]
> I'd prefer we include the link to the URL on gitignore.io that produced the ignore (for each) so it facilitates maintenance in the future.

I would add this to the two .gitginore. But frontend part is generated by the `create-vue-app`, So we don't have gitignore.io url.

> Also I have a very different yarn.lock because I'm on a different architecture.
> I also have a, not-on-repo but not ignored package-lock.json in frontend.
> I believe this could be solved by updating the documentation to use yarn install --frozen-lockfile instead of npm install as > > yarn resolver is more predictable and this option would not modify the lock file.

I used npm to create project. So the default document would be npm commands, I will update it then.

**Comment by rgaudin** [comment]
> Do I need to squash all the commits before merge, and which merge should I use? squash and merge or rebase?

Use merge commit. That's what we use because it retains the most information

**Pull request merged**